### DISCUSSION TITLE
Own map source

### DISCUSSION THREAD

**Discussion by ribanezu** [NREL/mappymatch#191] [Q&A]
Hi,
Great job with the repo! I wanted to ask if it's possible to use my own GeoJSON with the road network instead of using data from OSM.

**Comment by nreinicke** [comment]
Good question. We don't currently have first class support for easily loading your own map data in a GeoJSON format but you could probably do this by:

 1. Convert your map data into networkx ([maybe like this)](https://stackoverflow.com/a/70433839)
 2. Massage the networkx graph formatting so it matches what mappymatch expects:
 
![image](https://github.com/user-attachments/assets/0f84baf5-6579-4843-a655-b33b10ad21e0)

3. Load your networkx graph as an [`NxMap`](https://github.com/NREL/mappymatch/blob/142d810b2d338154605534d2bca0305f42ce23a3/mappymatch/maps/nx/nx_map.py#L41) and use it in mappymatch 

Apologies that we don't have this baked in a feature but we would always welcome a PR if you end up going down this path 😃
